// @capacitor/assets/dist/platforms/ios/assets.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IOS_3X_UNIVERSAL_ANYANY_SPLASH_DARK = exports.IOS_2X_UNIVERSAL_ANYANY_SPLASH_DARK = exports.IOS_1X_UNIVERSAL_ANYANY_SPLASH_DARK = exports.IOS_3X_UNIVERSAL_ANYANY_SPLASH = exports.IOS_2X_UNIVERSAL_ANYANY_SPLASH = exports.IOS_1X_UNIVERSAL_ANYANY_SPLASH = exports.IOS_1024_ICON = exports.IOS_83_5_PT_2X_ICON = exports.IOS_76_PT_2X_ICON = exports.IOS_76_PT_1X_ICON = exports.IOS_60_PT_3X_ICON = exports.IOS_60_PT_2X_ICON = exports.IOS_40_PT_3X_ICON = exports.IOS_40_PT_2X_1_ICON = exports.IOS_40_PT_2X_ICON = exports.IOS_40_PT_ICON = exports.IOS_29_PT_3X_ICON = exports.IOS_29_PT_2X_1_ICON = exports.IOS_29_PT_2X_ICON = exports.IOS_29_PT_ICON = exports.IOS_20_PT_3X_ICON = exports.IOS_20_PT_2X_1_ICON = exports.IOS_20_PT_2X_ICON = exports.IOS_20_PT_ICON = void 0;
/**
 * 20pt Icon
 *
 * - iPhone Notification (iOS 7-15)
 * - iPad Notification (iOS 7-15)
 */
exports.IOS_20_PT_ICON = {
    platform: "ios" /* Ios */,
    idiom: "ipad" /* iPad */,
    kind: "notification-icon" /* NotificationIcon */,
    name: 'AppIcon-20x20@1x.png',
    format: "png" /* Png */,
    width: 20,
    height: 20,
    scale: 1,
};
exports.IOS_20_PT_2X_ICON = {
    platform: "ios" /* Ios */,
    idiom: "iphone" /* iPhone */,
    kind: "notification-icon" /* NotificationIcon */,
    name: 'AppIcon-20x20@2x.png',
    format: "png" /* Png */,
    width: 40,
    height: 40,
    scale: 2,
};
exports.IOS_20_PT_2X_1_ICON = {
    platform: "ios" /* Ios */,
    idiom: "ipad" /* iPad */,
    kind: "notification-icon" /* NotificationIcon */,
    name: 'AppIcon-20x20@2x-1.png',
    format: "png" /* Png */,
    width: 40,
    height: 40,
    scale: 2,
};
exports.IOS_20_PT_3X_ICON = {
    platform: "ios" /* Ios */,
    idiom: "iphone" /* iPhone */,
    kind: "notification-icon" /* NotificationIcon */,
    name: 'AppIcon-20x20@3x.png',
    format: "png" /* Png */,
    width: 60,
    height: 60,
    scale: 3,
};
/**
 * 29pt Icon
 *
 * - iPhone Settings (iOS 7-15)
 * - iPad Settings (iOS 7-15)
 * - Apple Watch Companion Settings
 * - Apple Watch Notification Center
 */
exports.IOS_29_PT_ICON = {
    platform: "ios" /* Ios */,
    idiom: "ipad" /* iPad */,
    kind: "settings-icon" /* SettingsIcon */,
    name: 'AppIcon-29x29@1x.png',
    format: "png" /* Png */,
    width: 29,
    height: 29,
    scale: 1,
};
exports.IOS_29_PT_2X_ICON = {
    platform: "ios" /* Ios */,
    idiom: "iphone" /* iPhone */,
    kind: "settings-icon" /* SettingsIcon */,
    name: 'AppIcon-29x29@2x.png',
    format: "png" /* Png */,
    width: 58,
    height: 58,
    scale: 2,
};
exports.IOS_29_PT_2X_1_ICON = {
    platform: "ios" /* Ios */,
    idiom: "ipad" /* iPad */,
    kind: "settings-icon" /* SettingsIcon */,
    name: 'AppIcon-29x29@2x-1.png',
    format: "png" /* Png */,
    width: 58,
    height: 58,
    scale: 2,
};
exports.IOS_29_PT_3X_ICON = {
    platform: "ios" /* Ios */,
    idiom: "iphone" /* iPhone */,
    kind: "settings-icon" /* SettingsIcon */,
    name: 'AppIcon-29x29@3x.png',
    format: "png" /* Png */,
    width: 87,
    height: 87,
    scale: 3,
};
/**
 * 40pt Icon
 *
 * - iPhone Spotlight (iOS 7-15)
 * - iPad Spotlight (iOS 7-15)
 * - Apple Watch Home Screen
 */
exports.IOS_40_PT_ICON = {
    platform: "ios" /* Ios */,
    idiom: "ipad" /* iPad */,
    kind: "spotlight-icon" /* SpotlightIcon */,
    name: 'AppIcon-40x40@1x.png',
    format: "png" /* Png */,
    width: 40,
    height: 40,
    scale: 1,
};
exports.IOS_40_PT_2X_ICON = {
    platform: "ios" /* Ios */,
    idiom: "iphone" /* iPhone */,
    kind: "spotlight-icon" /* SpotlightIcon */,
    name: 'AppIcon-40x40@2x.png',
    format: "png" /* Png */,
    width: 80,
    height: 80,
    scale: 2,
};
exports.IOS_40_PT_2X_1_ICON = {
    platform: "ios" /* Ios */,
    idiom: "ipad" /* iPad */,
    kind: "spotlight-icon" /* SpotlightIcon */,
    name: 'AppIcon-40x40@2x-1.png',
    format: "png" /* Png */,
    width: 80,
    height: 80,
    scale: 2,
};
exports.IOS_40_PT_3X_ICON = {
    platform: "ios" /* Ios */,
    idiom: "iphone" /* iPhone */,
    kind: "spotlight-icon" /* SpotlightIcon */,
    name: 'AppIcon-40x40@3x.png',
    format: "png" /* Png */,
    width: 120,
    height: 120,
    scale: 3,
};
/**
 * 60pt Icon
 *
 * - iPhone App (iOS 7-15)
 */
exports.IOS_60_PT_2X_ICON = {
    platform: "ios" /* Ios */,
    idiom: "iphone" /* iPhone */,
    kind: "icon" /* Icon */,
    name: 'AppIcon-60x60@2x.png',
    format: "png" /* Png */,
    width: 120,
    height: 120,
    scale: 2,
};
exports.IOS_60_PT_3X_ICON = {
    platform: "ios" /* Ios */,
    idiom: "iphone" /* iPhone */,
    kind: "icon" /* Icon */,
    name: 'AppIcon-60x60@3x.png',
    format: "png" /* Png */,
    width: 180,
    height: 180,
    scale: 3,
};
/**
 * 76pt Icon
 *
 * - iPad App (iOS 7+)
 */
exports.IOS_76_PT_1X_ICON = {
    platform: "ios" /* Ios */,
    idiom: "ipad" /* iPad */,
    kind: "icon" /* Icon */,
    name: 'AppIcon-76x76@1x.png',
    format: "png" /* Png */,
    width: 76,
    height: 76,
    scale: 1,
};
exports.IOS_76_PT_2X_ICON = {
    platform: "ios" /* Ios */,
    idiom: "ipad" /* iPad */,
    kind: "icon" /* Icon */,
    name: 'AppIcon-76x76@2x.png',
    format: "png" /* Png */,
    width: 152,
    height: 152,
    scale: 2,
};
/**
 * 83.5pt Icon
 *
 * iPad Pro (12.9-inch)
 */
exports.IOS_83_5_PT_2X_ICON = {
    platform: "ios" /* Ios */,
    idiom: "ipad" /* iPad */,
    kind: "icon" /* Icon */,
    name: 'AppIcon-83.5x83.5@2x.png',
    format: "png" /* Png */,
    width: 167,
    height: 167,
    scale: 2,
};
/**
 * 1024px Icon
 *
 * - App Store
 */
exports.IOS_1024_ICON = {
    platform: "ios" /* Ios */,
    idiom: "ios-marketing" /* iOSMarketing */,
    kind: "icon" /* Icon */,
    name: 'AppIcon-512@2x.png',
    format: "png" /* Png */,
    width: 1024,
    height: 1024,
    scale: 1,
};
exports.IOS_1X_UNIVERSAL_ANYANY_SPLASH = {
    platform: "ios" /* Ios */,
    idiom: "universal" /* Universal */,
    kind: "splash" /* Splash */,
    name: 'Default@1x~universal~anyany.png',
    format: "png" /* Png */,
    width: 2732,
    height: 2732,
    orientation: "portrait" /* Portrait */,
    scale: 1,
    theme: "any" /* Any */,
};
exports.IOS_2X_UNIVERSAL_ANYANY_SPLASH = {
    platform: "ios" /* Ios */,
    idiom: "universal" /* Universal */,
    kind: "splash" /* Splash */,
    name: 'Default@2x~universal~anyany.png',
    format: "png" /* Png */,
    width: 2732,
    height: 2732,
    orientation: "portrait" /* Portrait */,
    scale: 2,
    theme: "any" /* Any */,
};
exports.IOS_3X_UNIVERSAL_ANYANY_SPLASH = {
    platform: "ios" /* Ios */,
    idiom: "universal" /* Universal */,
    kind: "splash" /* Splash */,
    name: 'Default@3x~universal~anyany.png',
    format: "png" /* Png */,
    width: 2732,
    height: 2732,
    orientation: "portrait" /* Portrait */,
    scale: 3,
    theme: "any" /* Any */,
};
exports.IOS_1X_UNIVERSAL_ANYANY_SPLASH_DARK = {
    platform: "ios" /* Ios */,
    idiom: "universal" /* Universal */,
    kind: "splash-dark" /* SplashDark */,
    name: 'Default@1x~universal~anyany-dark.png',
    format: "png" /* Png */,
    width: 2732,
    height: 2732,
    orientation: "portrait" /* Portrait */,
    scale: 1,
    theme: "dark" /* Dark */,
};
exports.IOS_2X_UNIVERSAL_ANYANY_SPLASH_DARK = {
    platform: "ios" /* Ios */,
    idiom: "universal" /* Universal */,
    kind: "splash-dark" /* SplashDark */,
    name: 'Default@2x~universal~anyany-dark.png',
    format: "png" /* Png */,
    width: 2732,
    height: 2732,
    orientation: "portrait" /* Portrait */,
    scale: 2,
    theme: "dark" /* Dark */,
};
exports.IOS_3X_UNIVERSAL_ANYANY_SPLASH_DARK = {
    platform: "ios" /* Ios */,
    idiom: "universal" /* Universal */,
    kind: "splash-dark" /* SplashDark */,
    name: 'Default@3x~universal~anyany-dark.png',
    format: "png" /* Png */,
    width: 2732,
    height: 2732,
    orientation: "portrait" /* Portrait */,
    scale: 3,
    theme: "dark" /* Dark */,
};
